libs/server/Storage/Session/MainStore/MainStoreOps.cs

```diff
@@ -636,7 +689,71 @@ private unsafe GarnetStatus RENAME(ArgSlice oldKeySlice, ArgSlice newKeySlice, S
                                     SET(ref newKey, ref value, ref context);
                                 }
                             }
+                            else if (
+                                (expireTimeMs == -1 || expireTimeMs > 0) &&
+                                    hasEtag)
+                            {
+                                // IsNX means if the newKey Exists do not set it
+                                // We can't use SET with not exist here so instead we will do an Exists and skip the seeting if exists 
+                                if (isNX && EXISTS(newKeySlice, storeType, ref context, ref objectContext) == GarnetStatus.OK)
+                                {
+                                    result = 0;
+                                    returnStatus = GarnetStatus.OK;
+                                    // Skip setting the new key and go to calling the part after that
+                                    goto AFTERNEWKEYSET;
+                                }
+
+                                if (isNX)
```

Comment:
consolidate this second if isNX comparison into the preceding block

Reply:
rewrote this to not need it at all